### PR TITLE
conversation list

### PR DESCRIPTION
**Pull request by bob2402**

### CODE REVIEW

File: app/UI/new-nav.tsx [hunk 1]
```diff
@@ -331,7 +331,7 @@ class GroupChatList extends Component<GroupChatListProps> {
     }
 }
 
-type func_GetConversationList = () => Set<PublicKey>;
```
Comment (BlowaterNostr) [inline app/UI/new-nav.tsx:334]: The previous return type makes no sense, good catch.

**Review by BlowaterNostr — COMMENTED** [review]
keep going

File: app/UI/conversation-list.ts [hunk 2]
```diff
@@ -18,7 +19,10 @@ export class DM_List implements ConversationListRetriever, NewMessageChecker, Us
 
     constructor(
         public readonly ctx: NostrAccountContext,
-    ) {}
+        private readonly relayRecordGetter: RelayRecordGetter,
+    ) {
+        this.getConversationList = this.getConversationList.bind(this);
```
Comment (BlowaterNostr) [inline app/UI/conversation-list.ts:24]: not a good practice

File: app/UI/conversation-list.ts [hunk 3]
```diff
@@ -18,6 +19,7 @@ export class DM_List implements ConversationListRetriever, NewMessageChecker, Us
 
     constructor(
         public readonly ctx: NostrAccountContext,
+        private readonly relayRecordGetter: RelayRecordGetter,
```
Comment (BlowaterNostr) [inline app/UI/conversation-list.ts:22]: This is bad because it introduced unnecessary coupling and dependency

**Review by BlowaterNostr — COMMENTED** [review]
@bob2402 It's possible to merge at this stage but the code quality actually needs improvements.

File: app/UI/conversation-list.ts (hunk 3 above)
Comment (bob2402) [reply app/UI/conversation-list.ts:22]: I try to improve it.

**Pull request merged**